company-box.el

```diff
@@ -839,13 +840,17 @@ It doesn't nothing if a font icon is used."
           (frame (company-box--get-frame (frame-parent)))
           (window (frame-local-getq company-box-window (frame-parent)))
           (char-width (frame-char-width frame))
+          (current-width (* (frame-width frame) char-width))
           ((start . end) (company-box--get-start-end-for-width window win-start))
           (width (+ (company-box--calc-len (window-buffer window) start end char-width)
                     (if (and (eq company-box-scrollbar t) (company-box--scrollbar-p frame)) (* 2 char-width) 0)
                     char-width))
           (width (max (min width
                            (* company-tooltip-maximum-width char-width))
-                      (* company-tooltip-minimum-width char-width)))
+                      (* company-tooltip-minimum-width char-width)
+                      (if company-tooltip-width-grow-only
```

Comment:
@leodag Thanks I understand the issue now.

I would prefer to avoid calling `set-frame-width` because it's a bit an expensive function.
Could you please remove that `set-frame-width` and try the following change ? 

```suggestion
                      (if (and company-tooltip-width-grow-only (frame-visible-p frame))
```

It should address the issue you describe